jbpm-persistence/jbpm-persistence-jpa/src/test/java/org/jbpm/persistence/processinstance/objects/TestEventEmitter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.persistence.processinstance.objects;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jbpm.persistence.api.integration.EventCollection;
+import org.jbpm.persistence.api.integration.EventEmitter;
+import org.jbpm.persistence.api.integration.InstanceView;
+import org.jbpm.persistence.api.integration.base.BaseEventCollection;
+
+
+public class TestEventEmitter implements EventEmitter {
+
+
+    private static Set<InstanceView<?>> events = new LinkedHashSet<>();
+    
+    @Override
+    public void deliver(Collection<InstanceView<?>> data) {
```

Comment:
@mswiderski Could you specify a use case for this method? When do I want to implement it?

Reply:
Take a look at Jira where I added two links that should give you some answers

Reply:
Thanks, it helped a lot. I'll plan some jira's for integration tests.